src/components/value/boolean/index.test.tsx

```diff
@@ -0,0 +1,284 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import * as React from "react";
+import { fireEvent, render, waitFor } from "@testing-library/react";
+import * as SolidFns from "@inrupt/solid-client";
+import BooleanValue from "./index";
+
+const mockPredicate = `http://xmlns.com/foaf/0.1/nick`;
+const mockNick = "test nick value";
+
+const mockThing = SolidFns.addStringNoLocale(
+  SolidFns.createThing(),
+  mockPredicate,
+  mockNick
+);
+
+const mockDataset = SolidFns.setThing(
+  SolidFns.mockSolidDatasetFrom("https://some-interesting-value.com"),
+  mockThing
+);
+const mockDatasetWithResourceInfo = SolidFns.setThing(
+  SolidFns.mockContainerFrom("https://some-interesting-value.com/"),
+  mockThing
+);
+
+const savedDataset = SolidFns.createSolidDataset() as any;
+jest
+  .spyOn(SolidFns, "saveSolidDatasetAt")
+  .mockImplementation(() => savedDataset);
```

Comment:
**Edit:** Just realised that this was already there in the original tests, so... You can update it, but if it worked, I guess it can stay as is too. :shrug: 

The type error when you remove the `as any` here is admittedly _very_ obtuse, but it _is_ meaningful: specifically, at some point it says `is missing the following properties from type 'Promise<...`.

In other words, `saveSolidDatasetAt is an async function, so it should return a Promise. You can do that by wrapping it in `Promise.resolve()`, but even easier is to let Jest handle that for you like in the tests:

```suggestion
const savedDataset = SolidFns.createSolidDataset() as any;
jest
  .spyOn(SolidFns, "saveSolidDatasetAt")
  .mockResolvedValue(savedDataset);
```

Likewise for the other components, of course :slightly_smiling_face: 

Unfortunately that still won't let you remove the `as any`, because `saveSolidDatasetAt` expects a few more things to be inside that SolidDataset - specifically, metadata about where it was fetched from (which is simulated by `mockSolidDatasetFrom`) and a log of Things that were added/removed. You can solve that by defining it as follows, but given how opaque that is (simulating a solid-client implementation detail, really), I'd be OK with the `any` too:

```
const savedDataset = setThing(SolidFns.mockSolidDatasetFrom("https://arbitrary.pod/resource"), createThing());
```